src/interface/cli/mod.rs

```diff
@@ -0,0 +1,86 @@
+use clap::{App, Arg, SubCommand, crate_authors, crate_version};
+use crate::readers;
+use crate::writers;
+use crate::writers::dir_builder::DirectoryBuilder;
+use std::path::Path;
+use std::process::Command;
+
+pub fn parse_args() {
+    let app_name = "apigen";
+    let about = "Generates directories and skeletons for APIs from .toml input file";
+    let matches = App::new(app_name)
+        .author(crate_authors!())
+        .version(crate_version!())
+        .about(about)
+        .subcommand(
+            SubCommand::with_name("build")
+                .about("Generate directory and associated files from .toml")
+                .arg(Arg::with_name("file")
+                    .help("The .toml file to read from")
+                    .required(true)
+                    .index(1)))
+        .subcommand(
+            SubCommand::with_name("run")
+                .about("Execute \"cargo run\" in directory specified in .toml file")
+                .arg(Arg::with_name("file")
+                    .help("The .toml file containing the path in which to execute \"cargo run\"")
+                    .required(true)
+                    .index(1)))
+        .get_matches();
+
+    if let Some(matches) = matches.subcommand_matches("build") {
+        let file = matches.value_of("file").unwrap();
+        println!("Building {:?}", file);
+        build_api(file).unwrap();
+    } else if let Some(matches) = matches.subcommand_matches("run") {
+        let file = matches.value_of("file").unwrap();
+        println!("Running {:?}", file);
+        run_api(file).unwrap();
+    }
+}
+
+fn build_api(file: &str) -> std::io::Result<()>{
+    // read in toml and print it (for debug)
+    let filename = String::from(file);
+    let toml_reader = readers::parser::InputFileReader::from_file(&filename);

```

Comment:
`file` leads me to believe that there's a file object open behind it.

the `file` (which is actually a `&str`) gets converted into a `filename: String` just so that it can later be used as a `&String`.

Changing the `file` input parameter to `filename` would be easier to understand and cut out a whole conversion out.